File: 1-TwoSum/1-TwoSum.js
// Last updated: 2/3/2026, 9:45:00 PM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map1= new Map()
    for (let i =0;i<nums.length;i++){
        let diff=target-nums[i]
         if (map1.has(diff)){
            return [map1.get(diff),i]
        }
        map1.set(nums[i],i)
    }
     return []
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 2/3/2026, 9:44:56 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        map={}


        for k,v in enumerate(nums):
            diff=target - v

            if diff in map:
                return [map[diff],k]

            map[v]=k
        return []